receiver/memcachedreceiver/scraper.go

```diff
@@ -16,91 +16,246 @@ package memcachedreceiver
+				}
+			case "bytes_read":
+				attributes.Insert(metadata.A.Direction, pdata.NewAttributeValueString("received"))
+				if parsedV, ok := r.parseInt(k, v); ok {
+					r.addToIntMetric(network, attributes, parsedV)
+				}
+			case "bytes_written":
+				attributes.Insert(metadata.A.Direction, pdata.NewAttributeValueString("sent"))
+				if parsedV, ok := r.parseInt(k, v); ok {
+					r.addToIntMetric(network, attributes, parsedV)
+				}
 			case "get_hits":
-				addIntSum(ilm.Metrics(), metadata.M.MemcachedGetHits.Init, now, parseInt(v))
+				attributes.Insert(metadata.A.Operation, pdata.NewAttributeValueString("get"))
+				attributes.Insert(metadata.A.Type, pdata.NewAttributeValueString("hit"))
+				if parsedV, ok := r.parseInt(k, v); ok {
+					r.addToIntMetric(operationCount, attributes, parsedV)
+				}
 			case "get_misses":
-				addIntSum(ilm.Metrics(), metadata.M.MemcachedGetMisses.Init, now, parseInt(v))
+				attributes.Insert(metadata.A.Operation, pdata.NewAttributeValueString("get"))
+				attributes.Insert(metadata.A.Type, pdata.NewAttributeValueString("miss"))
+				if parsedV, ok := r.parseInt(k, v); ok {
+					r.addToIntMetric(operationCount, attributes, parsedV)
+				}
+			case "incr_hits":
+				attributes.Insert(metadata.A.Operation, pdata.NewAttributeValueString("increment"))
+				attributes.Insert(metadata.A.Type, pdata.NewAttributeValueString("hit"))
+				if parsedV, ok := r.parseInt(k, v); ok {
+					r.addToIntMetric(operationCount, attributes, parsedV)
+				}
+			case "incr_misses":
+				attributes.Insert(metadata.A.Operation, pdata.NewAttributeValueString("increment"))
+				attributes.Insert(metadata.A.Type, pdata.NewAttributeValueString("miss"))
+				if parsedV, ok := r.parseInt(k, v); ok {
+					r.addToIntMetric(operationCount, attributes, parsedV)
+				}
+			case "decr_hits":
+				attributes.Insert(metadata.A.Operation, pdata.NewAttributeValueString("decrement"))
+				attributes.Insert(metadata.A.Type, pdata.NewAttributeValueString("hit"))
+				if parsedV, ok := r.parseInt(k, v); ok {
+					r.addToIntMetric(operationCount, attributes, parsedV)
+				}
+			case "decr_misses":
+				attributes.Insert(metadata.A.Operation, pdata.NewAttributeValueString("decrement"))
+				attributes.Insert(metadata.A.Type, pdata.NewAttributeValueString("miss"))
+				if parsedV, ok := r.parseInt(k, v); ok {
+					r.addToIntMetric(operationCount, attributes, parsedV)
+				}
+			case "rusage_system":
+				attributes.Insert(metadata.A.State, pdata.NewAttributeValueString("system"))
+				if parsedV, ok := r.parseFloat(k, v); ok {
+					r.addToDoubleMetric(rUsage, attributes, parsedV)
+				}
+
+			case "rusage_user":
+				attributes.Insert(metadata.A.State, pdata.NewAttributeValueString("user"))
+				if parsedV, ok := r.parseFloat(k, v); ok {
+					r.addToDoubleMetric(rUsage, attributes, parsedV)
+				}
 			}
 		}
-	}
 
+		// Calculated Metrics
+		r.calculateHitRatio("increment", "incr_hits", "incr_misses", stats.Stats, hitRatio)
```

Comment:
This is a really blurry boundary. We shouldn't delegate operations to a function that are already handled a bunch of times in this function. These should look quite similar to the above metrics, with the only difference being that we need to isolate two values and that we have to compute with them.

Pull the values out of the map and parse them here. 
```go
attributes.Insert(metadata.A.Operation, pdata.NewAttributeValueString("increment"))
parsedHit, okHit := r.parseInt("incr_hits", stats.Stats["incr_hits"])
parsedMiss, okMiss := r.parseInt("incr_misses", stats.Stats["incr_misses"])
if okHit && okMiss {
	r.addToDoubleMetric(rUsage, attributes, hitRatio(parsedHit, parsedMiss))
}
```

Reply:
Updated the handling, Although, the reasoning behind seperating it out is because instead of just looping through keys we are explicitly stating keys and each one needed two variables to be explicitly stated but I can see how it seems odd to seperate when we are doing similar operations above.